packages/gantt/frontend/sirius-components-gantt/src/helper/helper.tsx

```diff
@@ -180,3 +189,87 @@ export const getDisplayedColumns = (gqlColumns: GQLColumn[]): Column[] => {
 export const getSelectedColumns = (gqlColumns: GQLColumn[]) => {
   return gqlColumns.filter((col) => col.displayed === true).map((col) => TaskListColumnEnum[col.id]);
 };
+
+const getDayOfTheYear = (date: Date) => {
+  const start = new Date(date.getFullYear(), 0, 0);
+  const diff = date.getTime() - start.getTime() + (start.getTimezoneOffset() - date.getTimezoneOffset()) * 60 * 1000;
+  const oneDay = 1000 * 60 * 60 * 24;
+  const day = Math.floor(diff / oneDay);
+  return day;
+};
+
+export const roundDate = (
+  date: Date,
+  dateExtremity: DateExtremity,
+  action: BarMoveAction,
+  dateMoveStep: String
+): Date => {
+  const regex = /^(\d+)([DHm])$/;
+  const matches = dateMoveStep.match(regex);
+
+  if (matches && matches[1]) {
+    let value = parseInt(matches[1], 10);
+    const dimension = matches[2];
+    const newdate = new Date(date);
+    if (dimension == 'D') {
+      let dayOfTheYear: number = getDayOfTheYear(newdate);
+      if (newdate.getMinutes() != 0 || newdate.getHours() != 0 || (dayOfTheYear - 1) % value > 0) {
+        newdate.setMinutes(0);
+        newdate.setHours(0);
+        if (dateExtremity == 'start' || action == 'move') {
+          // In case of move we need to round start and end date with the same direction (floor) so that the duration keeps unchanged
+          dayOfTheYear = Math.floor((dayOfTheYear - 1) / value) * value + 1; // OK
+        } else if (dateExtremity == 'end') {
+          dayOfTheYear = Math.ceil(dayOfTheYear / value) * value + 1; // OK
+        }
+        newdate.setMonth(0);
+        newdate.setDate(1);
```

Comment:
Does this `setDate` is usefull ? Another `setDate` is done just after

Reply:
yes because setDate behaves differently according to the current date